cuchem/cuchem/datasets/molecule_loaders.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+import pathlib
+
+import cudf
+
+logger = logging.getLogger(__name__)
+
+
+class GenericCSVDataset():
+    def __init__(self, name=None, properties_cols=None, index_col=None, index_selection=None, max_len=None, data_path=None, data=None):
+        self.name = name
+        self.properties_cols = properties_cols
+        self.index_col = index_col
+        self.index_selection = index_selection
+        self.max_len = max_len
+        self.data_path = data_path
+        self.data = data
+
+    def _load_csv(self, columns, length_column=None, return_remaining=True):
+        columns = [columns] if not isinstance(columns, list) else columns
+        data = cudf.read_csv(self.data_path)
+
+        if self.index_col:
+            data = data.set_index(self.index_col).sort_index()
+        else:
+            data.index.name = 'index'
+
+        if self.index_selection:
+            data = data.loc[self.index_selection]
+            
+        if self.max_len:
+            data = data[data[length_column] <= self.max_len]
+
+        cleaned_data = data[columns]
+
+        if return_remaining:
+            if length_column:
+                remain_columns = [x for x in data.columns if (x not in columns) & (x not in [length_column])]
+            else:
+                remain_columns = [x for x in data.columns if (x not in columns)]
+            other_data = data[remain_columns]
+        else:
+            other_data = None
+        return cleaned_data, other_data
+
+    def load(self, columns=['canonical_smiles'], length_columns=['length']):
+        self.data, _ = self._load_csv(columns, length_columns)
+
+
+class ChEMBL_Approved_Drugs_Physchem(GenericCSVDataset):
```

Comment:
Nitpic: Here class name is a mix of camel case and '_' separated. Generally, camel case is used for class names. We may want to relax the rule to use uppercase for abbreviations. I had mad this mistake and fixing it when I come across these mistakes.